modules/oe_media_webtools/src/Plugin/Field/FieldWidget/OpPublicationListIdWidget.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_media_webtools\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextfieldWidget;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Defines the 'OP Publication List ID' field widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_media_op_publication_lists_id",
+ *   label = @Translation("OP Publication List ID"),
+ *   field_types = {
+ *     "json"
+ *   }
+ * )
+ */
+class OpPublicationListIdWidget extends StringTextfieldWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element = parent::formElement($items, $delta, $element, $form, $form_state);
+
+    $element['value']['#type'] = 'textfield';
+    $element['value']['#description'] = 'Enter the widget id of the snippet generated in <a href="https://op.europa.eu/en/my-widgets" target="_blank">OP Website</a>.';
+    $element['value']['#element_validate'] = [
+      [get_called_class(), 'validateInteger'],
+    ];
+
```

Comment:
<img width="655" alt="Screenshot 2020-12-04 at 21 01 41" src="https://user-images.githubusercontent.com/11507647/101204628-2c76f880-3675-11eb-8e84-e059df3b80f7.png">

```suggestion
    $element['value']['#value'] = $this->transformJsonToId();
```
or something of sorts, we transform the value to json in massageFormValues so we need to transform it back to an id for the edit form.